File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/params/Expression.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.SicParameter import SicParameter
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Int32 import Int32

@PythonTaskInfo
@SicParameter(name="N", default_value="10")
class Expression(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in", data_type=Int32)
        self.out = OutputPort(name="out", data_type=Int32)

    def runner_start(self):
        self.n = int(self.run_expression(self.get_parameter('N')))

    def execute(self, grp):
        self.out.put_data(Int32(self.in_.get_data(Int32).values[0]*self.n))

File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/params/Expression.java
package tasks.params;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.SicParameters;
import com.systemincloud.modeler.tasks.javatask.api.data.Int32;
import com.systemincloud.modeler.tasks.javatask.api.annotations.SicParameter;

@JavaTaskInfo
@SicParameters({
	@SicParameter(name=Expression.N)
}) public class Expression extends JavaTask {

	protected static final String N = "N";

	@InputPortInfo(name = "In", dataType = Int32.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Int32.class)
	public OutputPort out;

	private int n;

	@Override
	public void runnerStart() {
		this.n = (Integer) runExpression(getParameter(N));
	};

	@Override
	public void execute(int grp) {
		out.putData(new Int32(in.getData(Int32.class).getValue()*n));
	}

}
